src/Core/Layer/GeometryLayer.js

```diff
@@ -0,0 +1,104 @@
+import { EventDispatcher } from 'three';
+import Picking from '../Picking';
+
+function GeometryLayer(id, object3d) {
+    if (!id) {
+        throw new Error('Missing id parameter (GeometryLayer must have a unique id defined)');
+    }
+    if (!object3d || !object3d.isObject3D) {
+        throw new Error('Missing/Invalid object3d parameter (must be a three.js Object3D instance)');
+    }
+
+    this._attachedLayers = [];
+    this.type = 'geometry';
+    this.protocol = 'tile';
+    this.visible = true;
```

Comment:
would visibility and opacity not be rendering/style options?

Reply:
Yes for `opacity`, no for `visibility` imho.